File: lib/py/uapi/internal/binary/Pack.py
def pack(value: object) -> object:
    from ...internal.binary.PackList import pack_list

    if isinstance(value, list):
        return pack_list(value)
    elif isinstance(value, dict):
        new_map = {}

        for key, val in value.items():
            new_map[key] = pack(val)

        return new_map
    else:
        return value

File: lib/ts/uapi/internal/binary/Pack.ts
import { packList } from './uapi/internal/binary/PackList';

export function pack(value: any): any {
    if (Array.isArray(value)) {
        return packList(value);
    } else if (typeof value === 'object' && value !== null) {
        const newMap: any = {};

        for (const key in value) {
            if (value.hasOwnProperty(key)) {
                newMap[key] = pack(value[key]);
            }
        }

        return newMap;
    } else {
        return value;
    }
}
